pkg/tasktesting/run.go

```diff
@@ -187,6 +188,9 @@ func WatchTaskRunUntilDone(t *testing.T, testOpts TestOpts, tr *tekton.TaskRun)
 		case tr := <-taskRunDone:
 			cancel()
 			return tr, collectedLogsBuffer, nil
+		case <-ctx.Done():
+			cancel()
```

Comment:
Do you still need to `cancel()` when it is done?

Reply:
@henninggross and I looked this up in _Learning Go_ (https://learning.oreilly.com/library/view/learning-go/9781492077206/ch12.html#context_timers):

> Any time you create a context that has an associated cancel function, you must call that cancel function when you are done processing, whether or not your processing ends in an error. If you do not, your program will leak resources (memory and goroutines) and eventually slow down or crash. There is no error if you call the cancel function more than once; any invocation after the first does nothing. The easiest way to make sure you call the cancel function is to use defer to invoke it right after the cancel function is returned.

Therefore, we changed it to have `cancel()` called through `defer` (seems to be the common pattern, c.f. e.g. https://go.dev/src/context/example_test.go)